webapp/channels/src/workers/sw.js

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// Update this every time code is changed.
+// This is needed to delete old cache entries.
+const cacheName = 'v1';
+const HTML_REQUEST = new Request('root.html');
+
+const putInCache = async (request, response) => {
+    const cache = await caches.open(cacheName);
+    if (request.destination === 'document') {
+        await cache.put(HTML_REQUEST, response);
+    } else {
+        await cache.put(request, response);
+    }
```

Comment:
I'm not sure why we don't use the request itself as a key for documents, putting everything under the same root.html key. Do we only serve one and only document?